### PR TITLE
#105 Using sat to migrate twice in a row leads to an error

### PR DESCRIPTION
**Pull request by adrianchivu**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-Server-Administration-Tool/pulls) open
- [X] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-Server-Administration-Tool/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### CODE REVIEW

File: Modules/Common/Events/LogoutAndLoginEvent.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace Common.Events
+{
+    /// <summary>
+    /// A message bus event used to signify that data source will be disconnected and reconnected
+    /// </summary>
+    public class LogoutAndLoginEvent
+    {
+        public ISession CurrentSession { get; set; }
+    }
+}
```
Comment (cozminvelciu) [inline Modules/Common/Events/LogoutAndLoginEvent.cs:37]: Missing newline at end of file.

File: Modules/Migration.Tests/LoginViewModelTestFixture.cs [hunk 2]
```diff
@@ -93,44 +128,15 @@ public void VerifyIfLoginFailed()
...
-            Assert.AreEqual(true, this.loginViewModel.Object.LoginSuccessfully);
-
-            // relogin
-
-            Assert.DoesNotThrowAsync(async () => await this.loginViewModel.Object.LoginCommand.ExecuteAsyncTask());
-            Assert.AreEqual(true, this.loginViewModel.Object.LoginSuccessfully);
-            Assert.DoesNotThrowAsync(async () => await this.loginViewModel.Object.ServerSession.Close());
-        }
     }
 }
```
Comment (cozminvelciu) [inline Modules/Migration.Tests/LoginViewModelTestFixture.cs:142]: Missing newline at end of file.

**Review by cozminvelciu — COMMENTED** [review]
The diff is also showing the changes from #102. We probably need to update the branch.

File: Modules/Common/ViewModels/LoginViewModel.cs [hunk 3]
```diff
@@ -404,6 +414,25 @@ private async Task ExecuteLogin()
             }
         }
 
+        /// <summary>
+        /// Executes login command
+        /// </summary>
+        /// <param name="currentSession"></param>
```
Comment (cozminvelciu) [inline Modules/Common/ViewModels/LoginViewModel.cs:420]: Missing XML docs.

File: Modules/Migration/Utils/Migration.cs [hunk 4]
```diff
@@ -306,7 +293,7 @@ public async Task<bool> ExportData()
             }
             finally
             {
-                await this.TargetSession.Close();
+                CDPMessageBus.Current.SendMessage(new LogoutAndLoginEvent {CurrentSession = this.TargetSession});
```
Comment (cozminvelciu) [inline Modules/Migration/Utils/Migration.cs:296]: Add whitespace after opening brace and before closing brace,

File: Modules/Migration/Utils/Migration.cs [hunk 5]
```diff
@@ -450,7 +437,7 @@ public async Task<bool> PackData(string migrationFile)
             }
             finally
             {
-                await this.SourceSession.Close();
+                CDPMessageBus.Current.SendMessage(new LogoutAndLoginEvent {CurrentSession = this.SourceSession});
```
Comment (cozminvelciu) [inline Modules/Migration/Utils/Migration.cs:440]: Add whitespace after opening brace and before closing brace,

File: Modules/Migration/ViewModels/MigrationViewModel.cs [hunk 6]
```diff
@@ -239,9 +238,14 @@ public MigrationViewModel()
         /// <summary>
         /// Trigger loading of the person migration.json file to the application
         /// </summary>
-        [ExcludeFromCodeCoverage]
+
```
Comment (cozminvelciu) [inline Modules/Migration/ViewModels/MigrationViewModel.cs:241]: Unneeded extra newline.

**Review by cozminvelciu — CHANGES_REQUESTED** [review]

File: Modules/Common/ViewModels/LoginViewModel.cs [hunk 7]
```diff
@@ -417,7 +417,7 @@ private async Task ExecuteLogin()
         /// <summary>
         /// Executes login command
         /// </summary>
-        /// <param name="currentSession"></param>
+        /// <param name="currentSession">Current user session<see cref="ISession"/></param>
```
Comment (cozminvelciu) [inline Modules/Common/ViewModels/LoginViewModel.cs:420]: Missing whitespace. Would suggest:
```
/// <param name="currentSession">Current user <see cref="ISession"/></param>
```

File: Modules/Migration.Tests/LoginViewModelTestFixture.cs [hunk 8]
```diff
@@ -1,28 +1,31 @@
...
 //    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 //    Affero General Public License for more details.
-// 
+//
 //    You should have received a copy of the GNU Affero General Public License
 //    along with this program. If not, see <http://www.gnu.org/licenses/>.
 // </copyright>
 // --------------------------------------------------------------------------------------------------------------------
 
+using CDP4Dal;
```
Comment (cozminvelciu) [inline Modules/Migration.Tests/LoginViewModelTestFixture.cs:26]: Usings inside namespace.

**Review by cozminvelciu — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]

**Review by cozminvelciu — APPROVED** [review]

**Pull request merged**